src/modules/computerVision/gaborFilter/gaborFilter.tsx

```diff
@@ -0,0 +1,184 @@
+   let y1 = x.map((inner, i) => inner.map((v, j) => (-v * Math.sin(theta)) + (y[i][j] * Math.cos(theta))));
+   
+   let p = Math.pow(omega, 2) / (4* Math.PI * Math.pow(K, 2))
+   let gaussian = x1.map((inner, i) => inner.map((v, j) => p * Math.exp(-Math.pow(omega, 2) / (8*Math.pow(K, 2)) * ( 4 * Math.pow(v, 2) + Math.pow(y1[i][j], 2)) )));
+   let sinusoid = x1.map((inner, i) => inner.map((v, j) => (func(omega * v) * Math.exp(Math.pow(K, 2) / 2))));
+   let gabor = sinusoid.map((inner, i) => inner.map((v, j) => (v * gaussian[i][j])));
+
+   // console.log(gabor);
+   // console.log(gabor.flat());
+   return gabor;
+}
+// gaborFilter(4, 0.3, Math.PI/4)
+
+
+
+const GaborDemo = () => {
+    const [kernel, setKernel] = useState<number[] | undefined>(undefined);
+    const [kernelGrid, setKernelGrid] = useState<number[][] | undefined>(undefined);
+
+    const configureKernel = (kernelSize: number, 
+                             omega: number,
+                             theta: number,
+                             K: number) => {
+
+      //   const theta = Math.PI/4;
+        const func = Math.cos;
+      //   const K = Math.PI;
+        
+        const gabor = gaborFilter(kernelSize, omega, theta, func, K)
+        const newKernel = gabor.flat();
+        const newKernelGrid = gabor;
+        setKernel(newKernel);
+        setKernelGrid(newKernelGrid);
+    }
+
+    return (
+        <div className="m-4">
+            <KernelConfig onConfig={configureKernel}/>
+            <div className="m-4">
+                Gaussian Kernel
+                <table className="m-auto"><tbody>
+                    {kernelGrid?.map((row, i) => (
+                        <tr key={i}>
+                            {row.map((val, j) => (
+                                <td key={j}
+                                    style={getBg(val, kernel)}
+                                    className="border-2 border-charcoal p-2"
+                                    title={`${val}`}>
+                                    {val.toFixed(5)}
+                                </td>))}
+                        </tr>
+                    ))}
+                </tbody></table>
+            </div>
+            <InteractiveFilter kernel={kernel} imgUrl={jellyfish} />
+        </div>
+    )
+}
+
+const KernelConfig = (props: { onConfig: (kernelSize: number, omega: number, theta: number, K: number) => void }) => {
+    const [kernelSize, setKernelSize] = useState<number>(5);
+    const [omega, setOmega] = useState<number>(1);
+    const [theta, setTheta] = useState<number>(0);
+    const [K, setK] = useState<number>(3.14);
+
+    const changeOmega = (e: any) => setOmega(parseFloat(e.target.value));
+    const changeKernelSize = (e: any) => setKernelSize(parseInt(e.target.value));
+    const changeTheta = (e: any) => setTheta(parseFloat(e.target.value));
+    const changeK = (e: any) => setK(parseFloat(e.target.value));
+
+    const invalidSize = (kernelSize < 3 || kernelSize > 101)
```

Comment:
I'm guessing the kernel size is the only thing that needs validation, so we can probably use just one variable (instead of both `invalidSize` and `invalidConfig`).

Reply:
I set it to  < 1 and > 7 as invalid since thats what prof mingolla said (if I remember correctly from thursdays standup)